### PR TITLE
Make sure mem3_rep autocreates target shards

### PR DESCRIPTION
**Pull request by iilyak**
The change to our fancier history entries introduced a regression that
internal replication wouldn't automatically create the target shards.
This fixes the issue by adding a get_or_create_db/2 in mem3_rep and
switches the use of couch_db:open_int/2 to that function.

Conflicts:
	src/mem3_rpc.erl

### CODE REVIEW

File: src/mem3_rpc.erl [hunk 1]
```diff
@@ -275,6 +275,16 @@ rexi_call(Node, MFA) ->
     end.
 
 
+get_or_create_db(DbName, Options) ->
+    case couch_db:open_int(DbName, Options) of
+    {not_found, no_db_file} ->
+        twig:log(warn, "~p creating ~s", [?MODULE, DbName]),
```
Comment (nickva) [inline src/mem3_rpc.erl]: Don't we need couch_log here?

Comment (iilyak) [inline src/mem3_rpc.erl]: good catch

**Comment by iilyak** [comment]
@nickva: Commit is amended to use couch_log.

**Comment by nickva** [comment]
+1 

Verified by deleting a shard of a db on node1 00000000-1fffffff/rdyno_src_0001.1462912443.couch.
Then restarted cluster. And saw in node1's log this line:

`node1.log:2016-05-10 16:44:50.424 [warning] node1@127.0.0.1 <0.1897.0> fabric_rpc creating shards/00000000-1fffffff/rdyno_src_0001.1462912443`

And shard was re-created successfully.

File: src/mem3_rpc.erl [hunk 2]
```diff
@@ -275,6 +275,16 @@ rexi_call(Node, MFA) ->
     end.
 
 
+get_or_create_db(DbName, Options) ->
```
Comment (rnewson) [inline src/mem3_rpc.erl]: since this is a cut-and-paste of a live function in fabric, can we put some thought to keeping this DRY?

**Comment by iilyak** [comment]
New option `create_if_missing` is added to couch_db:open.
PRs:
- https://github.com/apache/couchdb-couch/pull/174
- https://github.com/apache/couchdb-fabric/pull/50

**Comment by iilyak** [comment]
Should I rollback cherry-pick and implement it as fresh commit to be polite?

**Comment by rnewson** [comment]
+1 but please remove the 'conflicts: mem3_rpc.erl' part of the commit before merging.

**Pull request merged**